src/hooks/tests/useMultiSelect.test.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { render, act } from "@testing-library/react";
```

Comment:
You mentioned that the `renderHook` and `act` were not working for you so you created a `setup` function. This does work but I think the issue is that the current `act` is coming from this package instead of the hook package. If you update the reference, it'll work. Because you don't need the "react" import anymore, update the name of the file from `.tsx` to `.ts`.

```
import { renderHook, act } from "@testing-library/react-hooks";
import useMultiSelect from "../useMultiSelect";

// remove `function setup ...`

// ...
it("should update the selectedItems when the onChange function is called", () => {
    const { result } = renderHook(() =>
      useMultiSelect({ multiSelectId: multiSelectId, items })
    );
    // selectedItems starts empty
    expect(result.current.selectedItems).toEqual({});

    act(() => result.current.onChange(items[0].id));
    // Add correct id to items array
    expect(result.current.selectedItems).toEqual({
      "multiselect-hook-test-id": { items: ["dogs"] },
    });

    act(() => result.current.onChange(items[1].id));
    // Adds second item to the selectedItems
    expect(result.current.selectedItems).toEqual({
      "multiselect-hook-test-id": { items: ["dogs", "cats"] },
    });
  });
...
```

Reply:
That worked! I am curious how importing react has such an effect. I did had act imported from the same package as renderHook in my first attempt. Thank you for this tip!